src/lib.rs

```diff
@@ -79,19 +94,52 @@ impl<T: BitAlloc> BitAlloc for BitAllocCascade16<T> {
             None
         }
     }
-    fn alloc_contiguous(&mut self, size: usize, align_log2: usize) -> Option<usize> {
-        if let Some(base) = find_contiguous(self, Self::CAP, size, align_log2) {
-            self.remove(base..base + size);
-            Some(base)
-        } else {
-            None
+
+    fn alloc_contiguous(
+        &mut self,
+        base: Option<usize>,
+        size: usize,
+        align_log2: usize,
+    ) -> Option<usize> {
+        match base {
+            Some(base) => check_contiguous(self, base, Self::CAP, size, align_log2).then(|| {
+                self.remove(base..base + size);
+                base
+            }),
+            None => find_contiguous(self, Self::CAP, size, align_log2).inspect(|&base| {
+                self.remove(base..base + size);
+            }),
         }
     }
-    fn dealloc(&mut self, key: usize) {
+
+    fn dealloc(&mut self, key: usize) -> bool {
         let i = key / T::CAP;
-        self.sub[i].dealloc(key % T::CAP);
         self.bitset.set_bit(i, true);
+        self.sub[i].dealloc(key % T::CAP)
     }
+
+    fn dealloc_contiguous(&mut self, base: usize, size: usize) -> bool {
+        let mut success = true;
+        let Range { start, end } = base..base + size;
+        assert!(start <= end);
+        assert!(end <= Self::CAP);
```

Comment:
Returns `false` if the check is not passed?

Reply:
> Returns `false` if the check is not passed?

just abandon the `assertion` here, right?